incremental-font-transfer/src/uri_templates.rs

```diff
@@ -343,26 +378,26 @@ impl ByteInfo {
     }
 }
 
+// This macro generates the byte info array at compile time.
 const NUM_U8S: usize = u8::MAX as usize + 1;
-
-static BYTE_INFO_MAP: OnceLock<[ByteInfo; NUM_U8S]> = OnceLock::new();
-
-/// Returns a map of information about each possible u8 byte value.
-///
-/// See ByteInfo for more details.
-fn byte_info_map() -> &'static [ByteInfo; NUM_U8S] {
-    // TODO(garretrieger): populate a static array using a macro instead of runtime initialization.
-    // TODO(garretrieger): map only byte values 0 to 7F, everything after defaults to the same value.
-    // See: https://datatracker.ietf.org/doc/html/rfc6570#section-2.1
-    BYTE_INFO_MAP.get_or_init(|| {
-        let mut info: [ByteInfo; NUM_U8S] = [ByteInfo::default(); NUM_U8S];
-        for value in 0..=u8::MAX {
-            info[value as usize] = ByteInfo::new(value);
-        }
-        info
-    })
+macro_rules! generate_byte_info_array {
+    () => {{
+        const ARRAY: [ByteInfo; NUM_U8S] = {
+            let mut info = [ByteInfo::Invalid; NUM_U8S];
+            let mut i = 0;
+            while i < NUM_U8S {
+                info[i] = ByteInfo::new(i as u8);
+                i += 1;
+            }
+            info
+        };
+        ARRAY
+    }};
 }
 
+/// This maps each possiblue byte (u8) value to an enum which classifies how that value is handled during expansion.
+static BYTE_INFO_MAP: [ByteInfo; NUM_U8S] = generate_byte_info_array!();
+
```

Comment:
The macro isn't actually doing anything here (other than outputting the literal source code inside it). And in general, `macro_rules` macros can't/don't evaluate procedural code. This is working with just because of const evaluation.

```suggestion
static BYTE_INFO_MAP: [ByteInfo; NUM_U8S] = {
    let mut info = [ByteInfo::Invalid; NUM_U8S];
    let mut i = 0;
    while i < NUM_U8S {
        info[i] = ByteInfo::new(i as u8);
        i += 1;
    }
    info
};
```

Reply:
Ah good catch, fixed.